core/storage/btree.rs

```diff
@@ -128,13 +128,95 @@ impl BTreeCursor {
         Ok(CursorResult::Ok(cell_count == 0))
     }
 
+    fn get_prev_record(&mut self) -> Result<CursorResult<(Option<u64>, Option<OwnedRecord>)>> {
+        loop {
+            let mem_page_rc = self.stack.top();
+            let cell_idx = self.stack.current_index();
+
+            // moved to current page begin
+            // todo: find a better way to flag moved to end or begin of page
+            if cell_idx < 0 {
+                loop {
+                    if self.stack.current_index() > 0 {
+                        self.stack.retreat();
+                        break;
+                    }
+                    if self.stack.has_parent() {
+                        self.stack.pop();
+                    } else {
+                        // moved to begin of btree
+                        return Ok(CursorResult::Ok((None, None)));
+                    }
+                }
+                // continue to next loop to get record from the new page
+                continue;
+            }
+
+            let cell_idx = cell_idx as usize;
+            debug!("current id={} cell={}", mem_page_rc.borrow().id, cell_idx);
+            if mem_page_rc.borrow().is_locked() {
+                return Ok(CursorResult::IO);
+            }
+            if !mem_page_rc.borrow().is_loaded() {
+                self.pager.load_page(mem_page_rc.clone())?;
+                return Ok(CursorResult::IO);
+            }
+            let mem_page = mem_page_rc.borrow();
+            let contents = mem_page.contents.as_ref().unwrap();
+
+            let cell_count = contents.cell_count();
+            let cell_idx = if cell_idx >= cell_count {
+                self.stack.set_cell_index(cell_count - 1);
```

Comment:
```suggestion
                self.stack.retreat();
```

Reply:
sometimes current `cell_idx` is `cell_count + 1`